base_location_nuts/models/res_partner.py

```diff
@@ -0,0 +1,80 @@
+# -*- coding: utf-8 -*-
+# Python source code encoding : https://www.python.org/dev/peps/pep-0263/
+##############################################################################
+#
+#    OpenERP, Open Source Management Solution
+#    This module copyright :
+#        (c) 2015 Antiun Ingenieria, SL (Madrid, Spain, http://www.antiun.com)
+#                 Antonio Espinosa <antonioea@antiun.com>
+#
+#    This program is free software: you can redistribute it and/or modify
+#    it under the terms of the GNU Affero General Public License as
+#    published by the Free Software Foundation, either version 3 of the
+#    License, or (at your option) any later version.
+#
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+
+from openerp import models, fields, api
+from openerp.tools.translate import _
+import collections
+
+
+def dict_recursive_update(d, u):
+    for k, v in u.iteritems():
+        if isinstance(v, collections.Mapping):
+            r = dict_recursive_update(d.get(k, {}), v)
+            d[k] = r
+        else:
+            d[k] = u[k]
+    return d
+
+
+class ResPartner(models.Model):
+    _inherit = 'res.partner'
+
+    region = fields.Many2one(comodel_name='res.partner.nuts',
+                             string="Region")
+    substate = fields.Many2one(comodel_name='res.partner.nuts',
+                               string="Substate")
+    lbl_region = fields.Char(compute='_labels_get')
+    lbl_substate = fields.Char(compute='_labels_get')
+
+    @api.one
+    @api.depends('country_id')
```

Comment:
Sorry but I don't understand the meaning of this `api.depends` since your labels are computed by the translation tools.
Even if you change the country, your labels will stay the same since they depend of the language of the connected user. Maybe I miss something.